lib/connextclient/ConnextClient.ts

```diff
@@ -0,0 +1,491 @@
+      ) || 'connext';
+    if (this.isDisabled()) {
+      status = errors.CONNEXT_IS_DISABLED.message;
+    } else {
+      try {
+        version = (await this.getVersion()).version;
+        const connextChannels = await this.getChannels();
+        channels = {
+          active: connextChannels.filter(c => c.state === 'opened').length,
+          settled: connextChannels.filter(c => c.state === 'settled').length,
+          closed: connextChannels.filter(c => c.state === 'closed').length,
+        };
+        address = this.address;
+        if (channels.active <= 0) {
+          status = errors.CONNEXT_HAS_NO_ACTIVE_CHANNELS().message;
+        }
+      } catch (err) {
+        status = err.message;
+      }
+    }
+
+    return {
+      chain,
+      status,
+      channels,
+      address,
+      version,
+    };
+  }
+
+  /**
+   * Gets the connext version.
+   */
+  public getVersion = async (): Promise<ConnextVersion> => {
+    return ({} as any);
+  }
+
+  /**
+   * Gets info about a given connext payment client.
+   * @param token_address the token address for the network to which the client belongs
+   * @param channel_address the address of the client to query
+   */
+  public getChannel = async (
+    token_address: string,
+    channel_address: string,
+  ): Promise<Channel> => {
+    return ({} as any);
+  }
+
+  /**
+   * Gets info about all non-settled channels.
+   * @param token_address an optional parameter to specify channels belonging to the specified token network
+   */
+  public getChannels = async (token_address?: string): Promise<Channel[]> => {
+    return ({} as any);
+  }
+
+  public channelBalance = async (
+    currency?: string,
+  ): Promise<ChannelBalance> => {
+    const client = this.getConnextClient();
+    if (!currency) {
+      return { balance: 0, pendingOpenBalance: 0, inactiveBalance: 0 };
+    }
+
+    const tokenAddress = this.getTokenAddress(currency);
+
+    const freeBalance = await client.getFreeBalance(tokenAddress);
+
+    const pendingOpenBalance = this.unitConverter.unitsToAmount({
+      currency,
+      units: freeBalance[client.multisigAddress].toNumber(),
+    });
+
+    const inactiveBalance = 0;
+
+    const balance = pendingOpenBalance + inactiveBalance;
+
+    return {
+      balance,
+      pendingOpenBalance,
+      inactiveBalance,
+    };
+  }
+
+  public tradingLimits = async (): Promise<TradingLimits> => {
+    // assume MAX_AMOUNT since Connext will re-collaterize accordingly
+    return {
+      maxSell: MAX_AMOUNT,
+      maxBuy: MAX_AMOUNT,
+    };
+  }
+
+  /**
+   * Returns the balances available in wallet for a specified currency.
+   */
+  public walletBalance = async (_currency?: string): Promise<WalletBalance> => {
+    return { totalBalance: 0, confirmedBalance: 0, unconfirmedBalance: 0 };
+  }
+  /**
+   * Creates a payment client.
+   */
+  public openChannel = async (): Promise<void> => {
+    if (!this.client) {
+      await this.initConnextClient();
```

Comment:
I'm thinking rather than call this initialization logic, this may just not be relevant to connext (as with the closeChannel method) and can do nothing or throw an exception.